fuel-tx/src/transaction/policies.rs

```diff
@@ -0,0 +1,329 @@
+                values[policy_type.index()] = default_values[policy_type.index()];
+            }
+        }
+        values
+    }
+}
+
+impl Serialize for Policies {
+    fn size_static(&self) -> usize {
+        self.bits.bits().size_static()
+    }
+
+    fn size_dynamic(&self) -> usize {
+        self.bits.bits().count_ones() as usize * Word::MIN.size()
+    }
+
+    fn encode_static<O: Output + ?Sized>(&self, buffer: &mut O) -> Result<(), Error> {
+        self.bits.bits().encode_static(buffer)
+    }
+
+    fn encode_dynamic<O: Output + ?Sized>(&self, buffer: &mut O) -> Result<(), Error> {
+        for (value, bit) in self.values.iter().zip(PoliciesBits::all().iter()) {
+            if self.bits.contains(bit) {
+                value.encode(buffer)?;
+            }
+        }
+        Ok(())
+    }
+}
+
+impl Deserialize for Policies {
+    fn decode_static<I: Input + ?Sized>(buffer: &mut I) -> Result<Self, Error> {
+        let bits = u32::decode(buffer)?;
+        let bits = PoliciesBits::from_bits(bits)
+            .ok_or(Error::Unknown("Invalid policies bits"))?;
+        Ok(Self {
+            bits,
+            values: Default::default(),
+        })
+    }
+
+    fn decode_dynamic<I: Input + ?Sized>(&mut self, buffer: &mut I) -> Result<(), Error> {
+        for (index, bit) in PoliciesBits::all().iter().enumerate() {
+            if self.bits.contains(bit) {
+                self.values[index] = Word::decode(buffer)?;
+            }
+        }
+
+        if let Some(maturity) = self.get(PolicyType::Maturity) {
+            if maturity > u32::MAX as u64 {
+                return Err(Error::Unknown("The maturity in more than `u32::MAX`"))
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(feature = "random")]
+impl Distribution<Policies> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Policies {
+        let bits: u32 = rng.gen();
+        let bits = bits & PoliciesBits::all().bits();
+        let bits = PoliciesBits::from_bits(bits).expect("We checked that above");
+        let values = rng.gen();
+        let mut policies = Policies {
+            bits,
+            values: Policies::values_for_bitmask(bits, values),
+        };
+
+        if policies.get(PolicyType::Maturity).is_some() {
+            let maturity: u32 = rng.gen();
+            policies.set(PolicyType::Maturity, Some(maturity as u64));
+        }
+
+        policies
+    }
+}
+
+#[test]
+fn values_for_bitmask_works() {
+    const MAX_BITMASK: u32 = 1 << POLICIES_NUMBER;
+    const VALUES: [Word; POLICIES_NUMBER] = [0x1000001, 0x2000001, 0x3000001, 0x4000001];
+
+    let mut set = hashbrown::HashSet::new();
+
+    for bitmask in 0..MAX_BITMASK {
+        let bits =
+            PoliciesBits::from_bits(bitmask).expect("Should construct a valid bits");
+        set.insert(Policies::values_for_bitmask(bits, VALUES));
+    }
+    assert_eq!(set.len(), MAX_BITMASK as usize);
+}
+
+#[test]
+fn canonical() {
```

Comment:
I think it's generally suggested to only have one `assert` per test. I'm kinda flexible on that when the entire test is the same except for the `then` section. That's assuming that it's communicated well why you're checking multiple things.